readthedocs/organizations/filters.py

```diff
@@ -0,0 +1,65 @@
+"""Filters used in organization dashboard."""
+
+import structlog
+from django.db.models import F
+from django.forms.widgets import HiddenInput
+from django.utils.translation import gettext_lazy as _
+from django_filters import CharFilter, FilterSet, OrderingFilter
+
+log = structlog.get_logger(__name__)
+
+
+class OrganizationSortOrderingFilter(OrderingFilter):
+
+    """Organization list sort ordering django_filters filter."""
+
+    SORT_NAME = "name"
+    SORT_CREATE_DATE = "pub_date"
```

Comment:
If these filters will be for an APIv3 endpoint, we have normalized all these date fields as `created` and we should stick with that standardization.

Reply:
This is not for an API v3 filter and this field was still `pub_date` on the model:
https://github.com/readthedocs/readthedocs.org/blob/7d65c16d654a8e45345866214b606c7536b424de/readthedocs/organizations/models.py#L33

Reply:
Cool -- as long as we are not exposing `pub_date=` to the URL, 👍🏼

Reply:
The URL would include `?sort=pub_date`. This is not ideal, but I think if we're taking the stance that the filter attributes should match the modeling, it's better to be consistent.

Reply:
However, I will add that in my other filter PR, I was arguing for nicer user facing URLs over technically absolute attributes (`version=foo` over `version__slug=foo`). I think `?sort=created` is a nicer URL in the same way. But, I'm sticking to the feedback from my other PR for now. We should come back to the filters later if we want to consolidate on a different pattern.

Reply:
> The URL would include `?sort=pub_date`. This is not ideal, but I think if we're taking the stance that the filter attributes should match the modeling, it's better to be consistent.

This is what I want to avoid.

With APIv3 (and the introduction of `TimeStampedModel`) we decided to standardize these fields and always expose them to the user as `created` and `modified`. If we are exposing now `pub_date` we are going backwards and creating a new inconsistency here.

I'd like to expose this field as `created` here as well instead of `pub_date`.

Reply:
I agree the field is not well named, but this seems like a change for the modeling.

This field name is direct from the modeling, not APIv3, and the organization models don't seem to subclass TimeStampedModel. The field name is unfortunate, but the proper place to fix this is going to be on the modeling, instead of renaming the field just for this filter.

Reply:
Yeah, the change should be eventually on the model, but we can't do that easily. That's why we decided to do it incrementally when we can (inheriting from TimeStampedModel when possible), but all the new code we write that is exposed to users uses `created` -- the new field name.

Reply:
I think `?sort=created` is fine as well, but I also don't personally agree we need to be strict about naming fields/values on view filters (this is not the same as API filters). However, if the aim is going to be making our view filters all strictly match our modeling, then we should make our modeling the source of truth.

We have had various discussions around this topic, so I think there is a bigger discussion here still. But in my opinion, *the view filters are more an extension of templating than anything* and something very different from our API. So, many of the filters do not reflect our modeling in the same way as our API, as the primary goal was to make the UI approachable to users.

For example, the build state filter:

https://github.com/readthedocs/readthedocs.org/blob/713bdbd5f7b3fac48c6536f1f3ccea7cc0e0b3e8/readthedocs/builds/filters.py#L53-L63

Making these fields/values strictly match our modeling would make the options awkward. Fields and URL would be weird either directly in display or the resulting URLs would be unnecessarily weird.

For example, `state` in this example includes an option for `success=False`. The user doesn't care that our modeling has two fields for `state` and `success` here, or that there are many various individual states that imply the build is active. Nor do we want to expose filtering options like `?state=active&success=false`.

The filter is not strict to modeling, but instead aims to provide the clearest options to the user. Our modeling aims to provide *us* with flexibility, but is definitely not what we want surfaced to the average user.

So, I think view filters are a fine area to obfuscate technical modeling patterns away from users, especially where filters are mostly annotated queries anyways.

I won't yet change anything here, it was already merged anyways. But I do agree we should continue discussion here, as I feel we're straddling patterns and standards and not yet in agreement on how these filters work best.

Reply:
> I think `?sort=created` is fine as well, but I also don't personally agree we need to be strict about naming fields/values on view filters (this is not the same as API filters).

The main point is about consistency all over our new code. Using `pub_date` is going backwards in the migration/consistency/naming we have done already.

Anyways, it seems I won't make you change your mind on this, so we can continue talking another time.